addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -627,6 +627,11 @@ def _unlink_if_no_leaves(self):
         if any(allocation.holiday_status_id.requires_allocation == 'yes' and allocation.leaves_taken > 0 for allocation in self):
             raise UserError(_('You cannot delete an allocation request which has some validated leaves.'))
 
+    def copy(self, default=None):
```

Comment:
I think you should override copy_data to avoid writing on the records after their creation.

Reply:
Hello, the records are auto validated upon creation (cf https://github.com/odoo/odoo/blob/90c9a2e6f5348bc9f29f1619c2fc69fba0dae4bf/addons/hr_holidays/models/hr_leave_allocation.py#L578-L579) so I thought that we actually need to write on them after creation.

Let me know if there is a better way to do it.
Thanks.

Reply:
Ah yes, I get your point ! 

Thanks :)